File: frontend/src/shared/api/dtos/properties/property-details.dto.ts
import type { PropertyType} from "@/shared/types/properties.ts";
import type { PropertyStatus } from "@/shared/types/properties.ts";

export type PropertyDetailsDto = {
  id: string;
  title: string;
  description: string;
  address: string;
  city: string;
  price: number;
  type: PropertyType;
  status: PropertyStatus;
  bedrooms: number;
  bathrooms: number;
  area: number;
  mainImageUrl?: string;
  brokerId: string;
  createdAt?: string;
  // TODO: add UpdatedAt
  // TODO: add imagesUrls: string[] - both for front and back
}

File: backend/src/Application/DTOs/Properties/PropertyDetailsDto.cs
using RealEstate.Domain.Enums.Properties;

namespace RealEstate.Application.DTOs.Properties;

public sealed record PropertyDetailsDto(
    Guid Id,
    string Title,
    string? Description,
    string Address,
    string City,
    decimal Price,
    PropertyType Type,
    int Bedrooms,
    int Bathrooms,
    decimal Area,
    PropertyStatus Status,
    string? MainImageUrl,
    Guid BrokerId,
    DateTime CreatedAt
    // TODO: add UpdatedAt
);
